src/c++/perf_analyzer/request_rate_worker.h

```diff
@@ -85,25 +81,52 @@ class RequestRateWorker : public LoadWorker {
             sequence_length, start_sequence_id, sequence_id_range, curr_seq_id,
             distribution, wake_signal, wake_mutex, execute),
         thread_config_(thread_config), max_threads_(max_threads),
-        start_time_(start_time), schedule_(schedule),
-        gen_duration_(gen_duration)
+        start_time_(start_time)
   {
   }
 
   void Infer() override;
 
+  /// Provides schedule information, where the consumer should
+  /// loop through the provided schedule, and then every time it loops back to
+  /// the start add an additional amount equal to the provided schedule_duration
+  ///
+  void SetSchedule(
+      RateSchedule schedule,
+      std::chrono::nanoseconds schedule_duration) override
+  {
+    schedule_ = schedule;
+    schedule_duration_ = schedule_duration;
+
+    schedule_rounds_ = 0;
+    schedule_index_ = 0;
+  }
+
+ private:
```

Comment:
This access specifier is redundant. Just move these data members to below the original `private:`.